HelpAssistHer/help-assist-her
# Load Initial Data in App

## Description
The initial data that we load should be loaded earlier when the app loads. Right now, it's loaded when you go to the verification portal. But if you only go to the mini app, it's not loaded at all. This is fixed by loading the initial data when the app component mounts. This way, we have access to it no matter which "app" we go to (mini app or verification portal). This PR makes that change and cleans up the code where it was loading before.

I also added the Google Maps API key so that we can use it in the mini app (to geocode addresses that users type in)

## Test Plan
### Test the server changes
1. Make sure to add the Google Maps API key as an env variable (you can get it from Heroku)
Run `export VERIFICATION_PORTAL_GOOGLE_MAPS_KEY={api_key_here}` in the terminal and then start the server
1. Go to http://localhost:4000/api/initial-data and make sure both the `facebookAppId` and the `googleMapsApiKey` are returned

### Test the client changes
1. Go to http://localhost:4000/ and make sure the initial data loads
1. If you go to the Redux inspector (dev tools), you should see the action
<img width="223" alt="screen shot 2018-08-01 at 7 51 29 am" src="https://user-images.githubusercontent.com/6378435/43529260-c51eb1cc-955f-11e8-83b7-549fed19345f.png">

3. Click Verification Portal and make sure there is _not_ another action that looks like the one above
4. Click the Login/Logout button to make sure that functionality still works
